### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by barnstee**
New docs.

### CODE REVIEW

**Review by dacolgit — COMMENTED** [review]
I know it's just a visualization aspect, but I think it's better to have the same displaying format for Command Line argument of version '2.5' and '2.6 and above'.

-> Fixed.

**Comment by koepalex** [comment]
_Readme.md_
* **in IEC62541 OPC UA PubSub standard format.** 2.5 publisher (in this repo) don't support PubSub, first sentence could be confusing

**Review by koepalex — APPROVED** [review]

File: TelemetryFormatConfiguration.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+```
+    // The configuration settings file consists of two objects:
+    // 1) The 'Defaults' object, which defines defaults for the telemetry configuration
```
Comment (mregen) [inline TelemetryFormatConfiguration.md:4]: I'd move pics and dependent md files to a docs folder.

File: CommandLineArguments.md [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+                                       iothubmessagesize parameter controls when
+                                       telemetry is sent.
+                                       Default: '10'
+              --dc, --deviceconnectionstring=VALUE
+                                     if publisher is not able to register itself with
+                                       IoTHub, you can create a device with name <
+                                       applicationname> manually and pass in the
+                                       connectionstring of this device.
+                                       Default: none
+          -c, --connectionstring=VALUE
```
Comment (mregen) [inline CommandLineArguments.md:108]: formatting

File: CommandLineArguments.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+        Options:
+              --pf, --publishfile=VALUE
+                                     the filename to configure the nodes to publish.
+                                       Default: '/appdata/publishednodes.json'
+              --tc, --telemetryconfigfile=VALUE
+                                     the filename to configure the ingested telemetry
+                                       Default: ''
+          -s, --site=VALUE           the site OPC Publisher is working in. if specified
+                                       this domain is appended (delimited by a ':' to
+                                       the 'ApplicationURI' property when telemetry is
```
Comment (mregen) [inline CommandLineArguments.md:30]: formatting

File: CommandLineArguments.md [hunk 4]
```diff
@@ -0,0 +1,331 @@
+### OPC Publisher Command Line Arguments for Version 2.5 and below
+
+        Usage: opcpublisher.exe <applicationname> [<iothubconnectionstring>] [<options>]
+    
+        applicationname: the OPC UA application name to use, required
+                         The application name is also used to register the publisher under this name in the
+                         IoTHub device registry.
+    
+        iothubconnectionstring: the IoTHub owner connectionstring, optional. Typically you specify the IoTHub owner connectionstring 	 only on the first start of the application. The connectionstring will be encrypted and stored in the platforms certificiate 	 store.
```
Comment (mregen) [inline CommandLineArguments.md]: > only on the first start of the appli [](start = 134, length = 37)

The latter portion of the sentence is invisible in the preview here: [https://github.com/Azure/iot-edge-opc-publisher/blob/docs/CommandLineArguments.md](https://github.com/Azure/iot-edge-opc-publisher/blob/docs/CommandLineArguments.md)

File: CommandLineArguments.md [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+    	On subsequent calls it will be read from there and reused. If you specify the connectionstring on each start, the device 		which is created for the application in the IoTHub device registry will be removed and recreated each time.
```
Comment (mregen) [inline CommandLineArguments.md]: >which is created for the application in the IoTHub device registry will be removed and recreate [](start = 128, length = 95)

same as above

File: CommandLineArguments.md [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+                                       sent to IoTHub.
+                                       The value must follow the syntactical rules of a
+                                       DNS hostname.
+                                       Default: not set
+              --ic, --iotcentral     publisher will send OPC UA data in IoTCentral
+                                       compatible format (DisplayName of a node is used
+                                       as key, this key is the Field name in IoTCentral)
+                                       . you need to ensure that all DisplayName's are
+                                       unique. (Auto enables fetch display name)
+                                       Default: False
```
Comment (mregen) [inline CommandLineArguments.md:41]: >  Default: False [](start = 37, length = 16)

I'd align these values with the start of sentence, so they better stand out

File: CommandLineArguments.md [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+                                        allowed values: Amqp, Http1, Amqp_WebSocket_Only,
+                                         Amqp_Tcp_Only, Mqtt, Mqtt_WebSocket_Only, Mqtt_
+                                        Tcp_Only) or IoT EdgeHub (allowed values: Mqtt_
+                                        Tcp_Only, Amqp_Tcp_Only).
+                                        Default for IoTHub: Mqtt_WebSocket_Only
+                                        Default for IoT EdgeHub: Amqp_Tcp_Only
+              --ms, --iothubmessagesize=VALUE
+                                     the max size of a message which can be send to
+                                       IoTHub. when telemetry of this size is available
+                                       it will be sent.
```
Comment (mregen) [inline CommandLineArguments.md]: >will [](start = 42, length = 4)

in general I'm not a big fan of future tense in technical documentation

File: CommandLineArguments.md [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+               --ih, --iothubprotocol=VALUE
+                                      the protocol to use for communication with IoTHub (
+                                        allowed values: Amqp, Http1, Amqp_WebSocket_Only,
+                                         Amqp_Tcp_Only, Mqtt, Mqtt_WebSocket_Only, Mqtt_
+                                        Tcp_Only) or IoT EdgeHub (allowed values: Mqtt_
+                                        Tcp_Only, Amqp_Tcp_Only).
+                                        Default for IoTHub: Mqtt_WebSocket_Only
+                                        Default for IoT EdgeHub: Amqp_Tcp_Only
+              --ms, --iothubmessagesize=VALUE
+                                     the max size of a message which can be send to
```
Comment (mregen) [inline CommandLineArguments.md]: >send [](start = 76, length = 4)

sent

File: CommandLineArguments.md [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+                                       IoTHub. when telemetry of this size is available
```
Comment (mregen) [inline CommandLineArguments.md]: >when [](start = 47, length = 4)

When

File: CommandLineArguments.md [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+              --si, --iothubsendinterval=VALUE
+                                     the interval in seconds when telemetry should be
+                                       send to IoTHub. If 0, then only the
+                                       iothubmessagesize parameter controls when
+                                       telemetry is sent.
+                                       Default: '10'
+              --dc, --deviceconnectionstring=VALUE
+                                     if publisher is not able to register itself with
+                                       IoTHub, you can create a device with name <
+                                       applicationname> manually and pass in the
```
Comment (mregen) [inline CommandLineArguments.md:105]: poor line braek

File: CommandLineArguments.md [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+                                       IoTHub. when telemetry of this size is available
+                                       it will be sent.
+                                       0 will enforce immediate send when telemetry is
+                                       available
+                                       Min: 0
+                                       Max: 262144
+                                       Default: 262144
+              --si, --iothubsendinterval=VALUE
+                                     the interval in seconds when telemetry should be
+                                       send to IoTHub. If 0, then only the
```
Comment (mregen) [inline CommandLineArguments.md]: >send [](start = 39, length = 4)

sent?

File: CommandLineArguments.md [hunk 12]
```diff
@@ -0,0 +1,331 @@
...
+                                     update the CRL passed in as base64 string to the
+                                       corresponding cert store (trusted or trusted
+                                       issuer)
+              --uc, --updatecrlfile=VALUE
+                                     update the CRL passed in as file to the
+                                       corresponding cert store (trusted or trusted
+                                       issuer)
+              --rc, --removecert=VALUE
+                                     remove cert(s) with the given thumbprint(s) (
+                                       multiple thumbprints supported)
```
Comment (mregen) [inline CommandLineArguments.md]: how to specify multiple?

File: CommandLineArguments.md [hunk 13]
```diff
@@ -0,0 +1,331 @@
...
+                                       Default: X509Store
+              --dp, --devicecertstorepath=VALUE
+                                     the path of the iot device cert store
+                                       Default Default (depends on store type):
+                                       X509Store: 'My'
+                                       Directory: 'CertificateStores/IoTHub'
+          -i, --install              register OPC Publisher with IoTHub and then exits.
+                                       Default:  False
+          -h, --help                 show this message and exit
+              --st, --opcstacktracemask=VALUE
```
Comment (mregen) [inline CommandLineArguments.md:288]: formatting

File: CommandLineArguments.md [hunk 14]
```diff
@@ -0,0 +1,331 @@
...
+                                        the rejected cert store will always reside in a
+                                       directory.
+              --it, --issuercertstoretype=VALUE
+                                     ignored.
+                                        the trusted issuer cert store will always
+                                       reside in a directory.
+
+
+
+### OPC Publisher Command Line Arguments for Version 2.6 and above
```
Comment (mregen) [inline CommandLineArguments.md:315]: >on 2.6 a [](start = 50, length = 8)

why 2.6. options here?

**Review by mregen — CHANGES_REQUESTED** [review]
:clock1:

File: opcpublisher.sln [hunk 15]
```diff
@@ -10,9 +10,11 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 		.gitignore = .gitignore
 		.travis.yml = .travis.yml
 		appveyor.yml = appveyor.yml
+		CommandLineArguments.md = CommandLineArguments.md
```
Comment (mregen) [inline opcpublisher.sln:13]: >CommandLineArguments.md = CommandLineArguments.md [](start = 2, length = 49)

I'd not add doc and license files to solutions

File: TelemetryFormatConfiguration.md [hunk 16]
```diff
@@ -0,0 +1,187 @@
+```
+    // The configuration settings file consists of two objects:
```
Comment (mregen) [inline TelemetryFormatConfiguration.md:3]: >/ The configuration s [](start = 5, length = 21)

The page has no header / description

File: CommandLineArguments.md (hunk 4 above)
Comment (barnstee) [reply CommandLineArguments.md]: Strange, it renders correctly in my MD editor. Will fix.

**Review by barnstee — COMMENTED** [review]
> _Readme.md_
> 
> * **in IEC62541 OPC UA PubSub standard format.** 2.5 publisher (in this repo) don't support PubSub, first sentence could be confusing

Fixed.

**Pull request merged**